src/bin/rustup-init.rs

```diff
@@ -62,18 +60,7 @@ fn run_rustup_inner() -> Result<utils::ExitCode> {
     utils::current_dir()?;
     utils::current_exe()?;
 
-    // The name of arg0 determines how the program is going to behave
-    let arg0 = match process().var("RUSTUP_FORCE_ARG0") {
-        Ok(v) => Some(v),
-        Err(_) => process().args().next(),
-    }
-    .map(PathBuf::from);
-    let name = arg0
-        .as_ref()
-        .and_then(|a| a.file_stem())
-        .and_then(std::ffi::OsStr::to_str);
-
-    match name {
+    match process().name().as_deref() {
```

Comment:
While this cleanup looks useful, it's not part of the effect the commit message suggests.  Could you break out this refactor into a commit of its own, likely titled `executable name via process abstraction` or somesuch?

Reply:
Done. Sorry for the delay!